### PR TITLE
[fix](multi-catalog)support the max compute partition prune

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

1. max compute partition prune,
we just support filter mc partitions by '='，it can filter just one partition
to support multiple partition filter and range operator('>','<', '>='..), the partition prune should be supported.

2. add max compute row count cache and partitionValues cache

3. add max compute regression case

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run external

**Comment by wsjz** [comment]
run buildall

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java [hunk 1]
```diff
@@ -173,6 +173,11 @@ public void open() throws IOException {
                     .collect(Collectors.toSet());
             List<Column> maxComputeColumns = new ArrayList<>(readColumns);
             maxComputeColumns.removeIf(e -> partitionColumns.contains(e.getName()));
+            if (maxComputeColumns.isEmpty() && !partitionColumns.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:176]: If use count(*), maxComputeColumns can be empty，but partitionColumns not.
Consider renaming

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/MaxComputeExternalTable.java [hunk 2]
```diff
@@ -73,85 +72,105 @@ protected synchronized void makeSureInitialized() {
         super.makeSureInitialized();
         if (!objectCreated) {
             odpsTable = ((MaxComputeExternalCatalog) catalog).getClient().tables().get(name);
+            initTablePartitions();
             objectCreated = true;
         }
     }
 
+    public long getTotalRows() throws TunnelException {
+        // use for non-partitioned table
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/external/MaxComputeExternalTable.java:81]: partitioned table will read whole partition each split now

[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/external/MaxComputeExternalTable.java:81]: add comment

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by AshinGau — APPROVED** [review]

**Comment by AshinGau** [comment]
LGTM

**Pull request merged**